alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/model/ActionExecutionResult.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.model
+
+import com.amazon.opendistroforelasticsearch.alerting.elasticapi.instant
+import com.amazon.opendistroforelasticsearch.alerting.elasticapi.optionalTimeField
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.ToXContentObject
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.io.IOException
+import java.lang.IllegalStateException
+import java.time.Instant
+
+/**
+ * When an alert triggered, the trigger's actions will be executed.
+ * Action execution result records action throttle result and is a part of Alert.
+ */
+data class ActionExecutionResult(
+    val actionId: String,
+    val lastExecutionTime: Instant?,
+    val throttledCount: Int = 0
+) : ToXContentObject {
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        return builder.startObject()
+                .field(ACTION_ID_FIELD, actionId)
+                .optionalTimeField(LAST_EXECUTION_TIME_FIELD, lastExecutionTime)
+                .field(THROTTLED_COUNT_FIELD, throttledCount)
+                .endObject()
+    }
+
+    companion object {
+        const val ACTION_ID_FIELD = "action_id"
+        const val LAST_EXECUTION_TIME_FIELD = "last_execution_time"
+        const val THROTTLED_COUNT_FIELD = "throttled_count"
+
+        @JvmStatic
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser): ActionExecutionResult {
+            var actionId: String? = null
```

Comment:
I believe we have been using `lateinit var actionId: String` instead of `String?` elsewhere

Reply:
Thanks, will change it